### ISSUE TITLE
Docker-compose: MariaDB/MySQL communication issues

### ISSUE THREAD

**Issue by pablorq** [n8n-io/n8n#3138]
**Describe the bug**
In a docker-compose stack with 2 containers (n8n & db), the database (db) container receives incorrect packets from n8n app.

The error shown in the log is:
```
[Note] Aborted connection xx to db: 'n8n' user: 'root' host: 'x.x.x.x' (Got an error reading communication packets)
```

From the n8n app container log:
```
There was an error initializing DB: "connect ECONNREFUSED x.x.x.x:3306"
```

As a result, the n8n service doesn't work.

Tested with MariaDB and MySQL databases.

**To Reproduce**
1. Create a docker-compose file or use Portainer Editor (my case):
```
version: '3.1'

services:
  db:
    image: mariadb:10.7
    restart: unless-stopped
    environment:
      - MARIADB_ROOT_PASSWORD=mysecretpassword
      - MARIADB_DATABASE=n8n

  n8n:
    image: n8nio/n8n
    restart: unless-stopped
    ports:
      - 5678:5678
    environment:
      - DB_TYPE=mariadb
      - DB_MYSQLDB_HOST=db
      - DB_MYSQLDB_DATABASE=n8n
      - DB_MYSQLDB_USER=root
      - DB_MYSQLDB_PASSWORD=mysecretpassword

    depends_on:
      - db
```
2. Run the stack
3. Check logs

**Expected behavior**
It is to get the first run web page.

**Environment (please complete the following information):**
 - OS: Ubuntu Linux 21.10
 - n8n Version: 0.172.0
 - Database system: MariaDB, MySQL

**Comment by Joffcom** [comment]
Hey @pablorq,

This doesn't look to be an issue with n8n, I have taken your compose file and put it into portainer and checked the log and it looks like MariaDB restarts itself. What I have done to get it working in Portainer was remove the restart option then wait for n8n to stop because it can't connect and restart it manually once MariaDB is ready for connections and it works.

I am not sure why the restart always doesn't magically start working or why the depends_on isn't working as I would have expected it to that is more of a docker question but hopefully by following the above you will be able to get it working as well.

**Comment by luizeof** [comment]
@pablorq I don't know if it's related, but I've had problems with the N8N and mariadb... it ran fine with percona-server 8.

**Comment by luizeof** [comment]
@pablorq I don't know if it's related, but I've had problems with the N8N and mariadb... it ran fine with percona-server 8.

**Comment by netroy** [comment]
I think the issue here might be the missing health-check on MariaDB. 
Without the health-check, `n8n` container gets started before MariaDB is done initializing the database.

I've created [a PR](https://github.com/n8n-io/n8n/pull/3384) with a health-check.

@luizeof can you try the compose file from PR please?

**Comment by pablorq** [comment]
@Joffcom,

That's a problem, if you have to manually start the containers. Maybe it can be added on n8n a mechanism to wait until the DB is connected (several systems do that).


@netroy,

That's sound a good idea! I'm waiting to test that. :slightly_smiling_face:

**Comment by Joffcom** [comment]
@pablorq that is not a bad idea, A lot of software I have worked with in the past will keep trying the connection but from what I remember I don't think docker was playing ball either. In theory if it is restarting on error it should try the connection again on the next restart as it is a startup task.

**Comment by netroy** [comment]
@pablorq this docker-compose changes were released in `n8n@0.189.0`. Can you please test if this fixes the issue?
